src/commonMain/kotlin/baaahs/PubSub.kt

```diff
@@ -253,5 +294,14 @@ abstract class PubSub {
                 }
             }
         }
+
+        @JsName("addStateChangeListener")
+        fun addStateChangeListener(callback: () -> Unit) {
```

Comment:
We should probably add a `removeStateChangeListener` and fix `App.js` to remove itself appropriately.

Reply:
Could we remove all state change listeners right before we add a new one? I know currently we are dealing with an array of state change listeners but we will only ever need one right?

Reply:
I spose. It's kind of an antipattern though; only a single listener is allowed, old listeners get unceremoniously dropped, and because there's no point in listeners unregistering themselves we'll get dangling references to dead listeners.

Otoh it's a bit annoying to hang on to the listener just so we can unlisten later. I think it's probably the better option though, pattern-wise.

Reply:
Okay - that said, I'm not sure where we would unlisten or what that method would look like - can we look at this next time we are together?

Reply:
I think it would be sorta like [this](https://github.com/baaahs/sparklemotion/blob/master/src/jsMain/js/app/components/gadgets/Slider/index.jsx#L33) — you'd need to save off the listener function when you add it.